lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -678,6 +687,36 @@ func (c *ContainerProxy) State(vc *viccontainer.VicContainer) (*types.ContainerS
 	return inspectJSON.State, nil
 }
 
+// exitCode returns container exitCode
+func (c *ContainerProxy) exitCode(vc *viccontainer.VicContainer) (string, error) {
```

Comment:
@caglar10ur I think this code is essentially TaskInspect - please make sure your branch is correctly populating the exitCode in the eventMonitor.